lib/contracts/builtin_contracts.rb

```diff
@@ -405,6 +405,7 @@ def initialize(contract_hash)
 
       def valid?(arg)
         return false unless arg.is_a?(Hash)
+        return false unless arg.keys.sort.eql?(contract_hash.keys.sort)
 
         contract_hash.all? do |key, _v|
           contract_hash.key?(key) && Contract.valid?(arg[key], contract_hash[key])
```

Comment:
is there any point of accessing `contract_hash[key]` instead of using `_v` from the iterator? (which should be renamed to `contract` in such case)